client/galaxy/scripts/apps/analysis.js

```diff
@@ -83,6 +83,7 @@ window.app = function app( options, bootstrapped ){
             '(/)user(/)(:form_id)' : 'show_user_form',
             '(/)workflow(/)' : 'show_workflows',
             '(/)pages(/)(:action_id)' : 'show_pages',
+            '(/)datasets(/)(:action_id)' : 'show_datasets',
```

Comment:
I went ahead and merged this to keep moving along, but we don't use the :action_id at all in the routing function below, so `/datasets/foo/` goes to `/datasets/list/` as well, for example.